modules/300-prometheus/hooks/deprecate_outdated_grafana_settings.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2021 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnAfterHelm: &go_hook.OrderedConfig{Order: 10},
+	Queue:       "/modules/prometheus/deprecate_outdated_grafana_settings",
+}, grafanaSettingsHandler)
+
+func grafanaSettingsHandler(input *go_hook.HookInput) error {
+	customPlugins := input.Values.Get("prometheus.grafana.customPlugins")
+	if customPlugins.Exists() {
+		pluginList := customPlugins.Array()
+		for _, plugin := range pluginList {
+			input.MetricsCollector.Set("d8_grafana_settings_outdated_plugin",
```

Comment:
Does it mean that all plugins are outdated? It seems misleading. Outdated plugins are plugins that require an update. The alert is probably not about outdated plugins, but about the deprecation of the `customPlugins` option?

Reply:
Actually we didn't plan to deprecate this option, it's more like a suggestion to check the plugins that were installed for previous grafana revision, so they may be incompatible with grafana 10.  
The naming was not the best, i agree, been trying to pick the better name instead of `deprecation`, so i renamed the hook to `suggest_custom_plugins_check`

So right now there are following alerts:

[deprecate_outdated_grafana_dashboard_crd]
- GrafanaDashboardPanelIntervalDeprecated (autoresolved by schedule within an hour)
- GrafanaDashboardAlertRulesDeprecated (autoresolved by schedule within an hour)
- `GrafanaDashboardPluginsMayBeOutdated`* (autoresolved by schedule within an hour) (but may alert on plugins that are valid, just not grafana internal, so it's unclear when to stop firing them)

*it's about grafana plugins that are panels that were found using on clustered dashboards 

[suggest_custom_plugins_check.go]
- `GrafanaPluginsMayBeOutdated`* (no autoresolve, can be changed to use schedule with able to keep getting values from `prometheus.grafana.customPlugins`?) (may alert on plugins that are valid, just not grafana internal, so it's unclear when to stop firing them)

*it's about any plugins (can be a datasources) installed for grafana 

**[to reduce the amount of fired alerts]**
Is there any way we could provide user a flag to disable alerts for `GrafanaDashboardPluginsMayBeOutdated` and `GrafanaPluginsMayBeOutdated` (simple `return nil` inside the hook handler) without changing the spec (that is required to add fields like `prometheus.grafana.disableDeprecationAlerts`)?